### PR TITLE
follow @jennybc advice

### PR DESCRIPTION
**Pull request by pachadotdev**
this is a new version of https://github.com/r-lib/usethis/pull/1921

@jennybc  I took some time to type with 1 hand and I used the logic you proposed here https://github.com/r-lib/usethis/pull/1921#issuecomment-1941927554

### CODE REVIEW

File: R/cpp11.R [hunk 1]
```diff
@@ -9,9 +9,9 @@
 #'
 #' @export
 use_cpp11 <- function() {
-  check_installed("cpp11")
```
Comment (jennybc) [inline R/cpp11.R:12]: I think we want to keep this check.

File: R/cpp11.R [hunk 2]
```diff
@@ -24,6 +24,10 @@ use_cpp11 <- function() {
 
   check_cpp_register_deps()
 
+  roxygen_ns_append(
+    sprintf("@useDynLib %s, .registration = TRUE", project_name())
+  ) && roxygen_remind()
+
```
Comment (jennybc) [inline R/cpp11.R:30]: It appears that `use_src()` does exactly this already? Can you help me understand why `use_cpp11()` needs to do that again?

https://github.com/r-lib/usethis/blob/93f4410ab93cb6e19cd6a49cd4ce33f93055b38b/R/rcpp.R#L74-L75

File: tests/testthat/test-cpp11.R [hunk 3]
```diff
@@ -5,12 +5,14 @@ test_that("use_cpp11() requires a package", {
...
 test_that("use_cpp11() creates files/dirs, edits DESCRIPTION and .gitignore", {
+  skip_if_not_installed("cpp11")
+
   create_local_package()
   use_roxygen_md()
+  use_package_doc()
 
   local_interactive(FALSE)
   local_check_installed()
-  local_mocked_bindings(check_cpp_register_deps = function() invisible())
```
Comment (jennybc) [inline tests/testthat/test-cpp11.R:16]: In terms of skips and mocks, I would expect this to look very much like the same test for `use_rcpp()`:

https://github.com/r-lib/usethis/blob/a91943706708f7ab41105cc3d5edf6ee61bbfb15/tests/testthat/test-rcpp.R#L6-L21

Comment (pachadotdev) [reply tests/testthat/test-cpp11.R]: I changed it to reflect this. The test "use_rcpp_armadillo() creates Makevars..." is something I can implement later for cpp11armadillo.

**Comment by jennybc** [comment]
Whew, we got there!

**Pull request merged**